### ISSUE TITLE
Memory Leak Reintroduced

### ISSUE THREAD

**Issue by CShelton11** [OData/odata.net#1100]
Memory leak originally corrected under 16.4.0 has been re-introduced

### Assemblies affected

known to affect 7.4.0.

### Reproduce steps

Install 7.4.0 of client.  Create a loop where a large volume of data is retrieved from server.  memory leak will be present.  Virtual space will be consumed and out of memory exception will occur.

### Expected result

memory leak shouldn't exist

### Actual result

Virtual memory issue described in previous.

### Additional detail

*Optional, details of the root cause if known. Delete this section if you have no additional details to add.*

**Comment by emzo1981** [comment]
I Agree , we've notice memry leaks in latest dlls. After moving back to 6.14 its all fine. Please check this out , becopuse we're loosing few features that doesn't exist in 6.14

**Comment by kgjersda** [comment]
Have noticed huge memory usage....
